### PR TITLE
Add autonomous functions

### PR DESCRIPTION
**Pull request by JoshFarkas**
I added a function that moves forward for a set amount of time based on an argument, and a function that turns in a direction for a set amount of time. There is an unfinished function that turns for an inputted angle, but it needs a constant to convert angle to time.

### CODE REVIEW

File: robot.py [hunk 1]
```diff
@@ -8,14 +8,55 @@
 import wpilib
 from wpilib.drive import DifferentialDrive
 
+class DriveUtil:
+
+    def __init__(self, robot):
+        self.robot = robot
+        self.timer = wpilib.Timer
```
Comment (zoecarver) [inline robot.py:15]: `self.timer = wpilib.Timer()`

File: robot.py [hunk 2]
```diff
@@ -8,14 +8,55 @@
...
+        resetTimer()
+
+    def turnForTime(time, direction):
+        #Makes sure the directions are reversed while turning
```
Comment (zoecarver) [inline robot.py:19]: Space + period.

File: robot.py [hunk 3]
```diff
@@ -8,14 +8,55 @@
...
+            directionMod = 1
+
+        if self.timer.get() - self.startTime < time:
+            self.robot.tankDrive(.5 * directionMod, .5 * directionMod * -1)
+        else:
+            self.robot.tankDrive(0, 0)
+    
+    
+    def turnForAngle(angle):
+        turnTime = Abs(angle) #* Insert [angle -> time] CONSTANT here
```
Comment (zoecarver) [inline robot.py:32]: `abs` not `Abs`

Comment (zoecarver) [reply robot.py:32]: Is `#*` a todo? If so, we should just use `# TODO:` so its more clear.

File: robot.py [hunk 4]
```diff
@@ -8,14 +8,55 @@
...
+        #Makes sure the wheels are reversed while turning
```
Comment (zoecarver) [inline robot.py:33]: Space + period.

File: robot.py [hunk 5]
```diff
@@ -39,7 +80,7 @@ def teleopInit(self):
     def teleopPeriodic(self):
         """Runs the motors with tank steering"""
         self.myRobot.tankDrive(self.joystick.getRawAxis(1) * -1 * self.speed, self.joystick.getRawAxis(3) * -1 * self.speed)
-
+
```
Comment (zoecarver) [inline robot.py:83]: Whoops.

File: robot.py [hunk 6]
```diff
@@ -8,14 +8,55 @@
...
+    
+    def turnForAngle(angle):
+        turnTime = Abs(angle) #* Insert [angle -> time] CONSTANT here
+        #Makes sure the wheels are reversed while turning
+        if angle < 0:
+            directionMod = -1
+        else:
+            directionMod = 1
+
+        #Turns the robot while the timer is less than the time that the angle translates to
```
Comment (zoecarver) [inline robot.py:39]: Space before `Turns`

File: robot.py [hunk 7]
```diff
@@ -8,14 +8,55 @@
...
+        if self.timer.get() - self.startTime < turnTime:
+            #Moves the robot, wheels are reversed directions
```
Comment (zoecarver) [inline robot.py:41]: Ditto.

File: robot.py [hunk 8]
```diff
@@ -8,14 +8,55 @@
...
+            self.robot.tankDrive(.5 * directionMod, .5 * directionMod * -1)
+        else:
+            #Stops the robot
```
Comment (zoecarver) [inline robot.py:44]: Ditto

**Review by zoecarver — CHANGES_REQUESTED** [review]
Looks great. The logic is spot on. Very nice. 

A few comments, nothing too big, though.  Just small nits. 

Also, can you make sure this code works on the robot (update `autonomousPeriodic` to make it do something)?

**Comment by zoecarver** [comment]
Also, @KaviMD or I can help you rebase off master. Once you do this you will also get some of the auto examples I put there. 

Edit: you'll need to do that before you can merge (that's what conflicts indicate).

**Comment by JoshFarkas** [comment]
The #* was saying multiply by the constant, not a todo, also what does
space + period mean?

**Comment by zoecarver** [comment]
> The #* was saying multiply by the constant, not a todo

Ah, I see. That makes sense now. I was reading it too quickly, my bad. 
> also what does space + period mean?

I didn't phrase that very clearly. This is a nit-pick, it's not a big deal at all, but something that is a good habit. There should be a space between the comment delimiter (`#` in this case but the same goes for `//`) and the comment itself. The "period" part means that I think there should be a period at the end of the comment. This is more personal preference but, I find comments turn out better when they are formatted like sentences. If you don't want to do it that way, you don't have to.

**Pull request closed without merging**